### ISSUE TITLE
Receiving events for content after ad insertion

### ISSUE THREAD

**Issue by olgunkaya** [google/ExoPlayer#9401]
Hi,

I am updating the progress bar following receiving the Player's READY state event. But with ad insertion, I suppose this event **only** thrown once the preroll video ready and not after skipping the ad. 

How can I receive an event or detect the actual content is being played following the ad skipping action ?

exoplayer version we are on : 2.11.4 we are using the ima-extension not manual ad insertion.

**Comment by ojw28** [comment]
Have you read our [documentation on player events](https://exoplayer.dev/listening-to-player-events.html)? I think it probably covers this. Note specifically the section on playlist transitions.

**Comment by olgunkaya** [comment]
Well tbh, I read this and checked that MediaItem class as well. But couldn't find a proper way to understand getDuration returns the duration of actual content's duration.

**Comment by ojw28** [comment]
Am I right in thinking that you're interested in detecting when playback transitions from one item to another (e.g., because an ad ends or is skipped). In which case I think the relevant documentation is:

> Whenever the player changes to a new media item in the playlist onMediaItemTransition(MediaItem mediaItem, @MediaItemTransitionReason int reason) is called on registered Player.Listeners.

I'm unclear why duration was relevant, since that doesn't seem to be part of your initial question. It's also unclear on what kind of ad insertion you're doing, which may be relevant to someone answering this question.

**Comment by olgunkaya** [comment]
You are actually correct. I will close this one and dig more. Let me handle the  Exo 101 first and come here with more detailed question. Thanks for the help. The ad insertion is done through ima extension.

**Comment by ojw28** [comment]
No worries.

For duration, note that the player has both `getContentDuration()` and `getDuration()`. Similarly for position, the player has both `getContentPosition()` and `getCurrentPosition()`. These will normally return the same values as one another, but when you're playing an ad, the `getContent[...]` methods will give the values for the underlying content being played (i.e., not the ad) whereas the other methods will give the values for the ad. So, for example, you'll see `getCurrentPosition()` increasing as the ad plays back, whilst `getContentPosition()` will remain stationary and equal to the position of the ad break in the underlying content. You can query whether an ad is playing using `isPlayingAd()`.

There's quite a lot of detail in the [Player Javadoc](https://exoplayer.dev/doc/reference/com/google/android/exoplayer2/Player.html#getContentBufferedPosition()) about querying various properties during ad playback, which might also be helpful.

**Comment by olgunkaya** [comment]
Thanks for this much detailed explanation :) I found my silly bug. Actually the bug inherited from the forked library :) it's not a bug there in the original repo but when I got it into my business and add/update some ad insertion logic it becomes a silly bug caused by me :) 

Below code called when state changed to READY. loadVideoStarted property set and event triggered back to react native context. 
So, preroll started. State READY triggered for the ad and  above property set to false. 
Skip initiated for the ad.
Actual content started. State READY triggered but since I did not reset the value on ad events the bug occurred and I come here to cry :)

ps. don't know why go this much detail. sorry x 2 :)

![image](https://user-images.githubusercontent.com/1777728/132680602-b9de0612-4156-4985-ada2-17cc56751263.png)
